### ISSUE TITLE
Running on device causes Lost connection to device. Exited on startup

### ISSUE THREAD

**Issue by Moncader** [Dart-Code/Dart-Code#4726]
**Describe the bug**
While attempting to start a Flutter application, running from VSCode _only_ sometimes (about a 50% chance) simply fails after the APK gets installed, the app's splash screen shows, and the VM service gets connected to with the message

```
Lost connection to device.
Exited
```

My entire development team is encountering this issue on multiple computers (all Mac) across multiple types of Android devices.

**To Reproduce**
1. Create a new flutter project via vscode or command line (flutter create) with any newer version of Flutter
2. Run it on an Android device in debug mode
3. About 50% of the time it fails

**Expected behavior**
The app should run in debug mode.

**Please complete the following information:**
 - Operating System and version: Mac OS 13.5.1
 - VS Code version: 1.81.1 (Universal)
 - Dart extension version: v3.73.20230905
 - Dart/Flutter SDK version: 3.1.0/3.13.0
 - Target device (if the issue relates to Flutter debugging): Certainly happens on Pixel 6 Pro.

**Comment by DanTup** [comment]
Could you try capturing debugging logs while reproducing this?

- Run the [**Dart: Capture Debugging Logs** command](https://dartcode.org/docs/logging/#capture-debugging-logs)
- Reproduce the issue
- Click **Cancel** on the logging notification to stop logging and open the log
- Remove anything sensitive from the log
- Attach the log here, in a [gist](https://gist.github.com/), or email to me at logs@dartcode.org.

Thanks!

**Comment by Moncader** [comment]
@DanTup 
https://gist.github.com/Moncader/6f5072d72679da2cf5d9f34b6aa78b83
Here you go.

**Comment by DanTup** [comment]
Thanks - unfortunately I can't see anything useful in the log. It seems the app starts and then all requests to the VM fail with "Service has disappeared".

Can you easily try this with a different USB cable (or if debugging over WiFi, switch to a cable) just to eliminate connectivity issues?

Otherwise, this will likely need filing at https://github.com/flutter/flutter/ as something seems to be failing between the Flutter tool and the device and this isn't likely something the VS Code extension will be able to fix. Thanks!

**Comment by Moncader** [comment]
@DanTup 
We have several people with the same issue across multiple Mac computers and different phones.
This also seems to only occur on newer Flutter versions as this does not occur in Flutter 3.7.

This also only occurs with vscode. Doing a `flutter run` from the CLI or using Android Studio never has this problem.

Of course, I can cross post this over in the Flutter issue tracker, but from my point of view (as just a user) this seems to be related to vscode as it is the only place this occurs.

If you think that it's possible for a Flutter-side issue to affect only vscode then yes, I can post it over on Flutter as well.

**Comment by DanTup** [comment]
I understand, but if downgrading Flutter fixes it that suggests whatever changed is in Flutter.

One possible candidate is that with newer Flutter SDKs you may be using the new debug adapters, and there could be a difference in them that's triggering this. Can you try:

- Press `F1` to open the command palette
- Run the **Open User Settings (JSON)** command
- Add the following to the JSON:
  `"dart.previewSdkDaps": false,`
- See whether the issue persists

Let me know if this appears to solve the issue. Disabling the new debug adapters is not a long-term fix, but it might help track down if the issue is a difference in the debug adapters rather than another part of Flutter.

Thanks!

**Comment by DanTup** [comment]
Also - could you confirm exactly which host machines and OSes, and what mobile devices you have seen this issue in? It may be difficult to troubleshoot remotely, so finding a way to repro it locally would help significantly.

**Comment by Moncader** [comment]
I tried the previewSdkDaps change and the issue still persists.

Also, and I'm sorry for the backtracking and I've updated the issue to reflect this... but I once again tested on Flutter 3.7.7, a version known to not have this issue since it was released (we have a product released on this version) replicated the issue as well after doing a debug -> quit -> debug -> quit 5 times (20% or so chance to reproduce it seems?).
So the issue seems to not be dependant on the Flutter version.

I also double checked once again if using the Flutter CLI (flutter run) had the same issue and it does not. Android Studio also does not. It's only VSCode.

Here are some of the host machine's info and Android device info:

- MacBook Pro 14-inch, 2021, Apple M1 Max, Mac OS Ventura 13.5.1 (22G90)
- MacBook Pro 13-inch, 2022, Apple M2, Mac OS Ventura 13.5.1 (22G90)
- MacBook Pro 13-inch, 2022, Apple M2, Mac OS Ventura 13.1

- Pixel 6 Pro, Android 13, Build Number TQ3A.230805.001
- Xperia 5 III, Android 13, Build Number 61.2.C.0.161
- Pixel 6a, Android 13, Build Number TQ3A.230805.001
